### DISCUSSION TITLE
Google pay button is not showing on stripe sheet (Apple pay working fine)

### DISCUSSION THREAD

**Discussion by rajuislamdev** [flutter-stripe/flutter_stripe#1683] [Q&A]
flutter version: 3.19.3
flutter_stripe: ^10.1.1
ext.kotlin_version = '1.9.22'
classpath 'com.android.tools.build:gradle:8.1.3'

W/WindowOnBackDispatcher(22380): sendCancelIfRunning: isInProgress=falsecallback=androidx.activity.OnBackPressedDispatcher$Api34Impl$createOnBackAnimationCallback$1@ea7f522
I/flutter (22380): inactive
D/AppCompatDelegate(22380): Checking for metadata for AppLocalesMetadataHolderService : Service not found
W/.laundryrunnerz(22380): Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (unsupported, reflection, allowed)
W/.laundryrunnerz(22380): Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (unsupported, reflection, allowed)
W/WindowOnBackDispatcher(22380): OnBackInvokedCallback is not enabled for the application.
W/WindowOnBackDispatcher(22380): Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
D/TrafficStats(22380): tagSocket(187) with statsTag=0xffffffff, statsUid=-1
W/.laundryrunnerz(22380): Method java.lang.Object androidx.compose.runtime.snapshots.SnapshotStateMap.mutate(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
W/.laundryrunnerz(22380): Common causes for lock verification issues are non-optimized dex code
W/.laundryrunnerz(22380): and incorrect proguard optimizations.
W/.laundryrunnerz(22380): Method void androidx.compose.runtime.snapshots.SnapshotStateMap.update(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
W/.laundryrunnerz(22380): Method boolean androidx.compose.runtime.snapshots.SnapshotStateMap.removeIf$runtime_release(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
D/TrafficStats(22380): tagSocket(171) with statsTag=0xffffffff, statsUid=-1
D/TrafficStats(22380): tagSocket(179) with statsTag=0xffffffff, statsUid=-1
D/TrafficStats(22380): tagSocket(186) with statsTag=0xffffffff, statsUid=-1
D/EGL_emulation(22380): app_time_stats: avg=12.17ms min=2.79ms max=72.01ms count=50
D/CompatibilityChangeReporter(22380): Compat change id reported: 171228096; UID 10187; state: ENABLED

**Comment by Riyadh-Herizi** [comment]
i'm having the same issue with all recent versions

```
await Stripe.instance.initPaymentSheet(
            paymentSheetParameters: SetupPaymentSheetParameters(
          customFlow: false,
          customerEphemeralKeySecret:
              clientSecretResponse.data?.customerEphemeralKeySecret,
          customerId: clientSecretResponse.data?.customerId,
          paymentIntentClientSecret:
              clientSecretResponse.data?.paymentIntentClientSecret,
          merchantDisplayName: product.productName,
          applePay: const PaymentSheetApplePay(
            merchantCountryCode: 'GB',
          ),
          googlePay: const PaymentSheetGooglePay(
            merchantCountryCode: 'GB',
            testEnv: true,
          ),
          appearance: PaymentSheetAppearance(
            primaryButton: PaymentSheetPrimaryButtonAppearance(
      shapes: const PaymentSheetPrimaryButtonShape(blurRadius: 10),
      colors: PaymentSheetPrimaryButtonTheme(
        light: PaymentSheetPrimaryButtonThemeColors(
          background: Theme.of(context).colorScheme.secondary,
          text: Theme.of(context).colorScheme.primary,
          border: Theme.of(context).colorScheme.secondary,
        ),
      ),
        ),
          ),
          style: ThemeMode.dark,
        ));

       await Stripe.instance.presentPaymentSheet();```

**Comment by captnCC** [comment]
It seems to be a general problem with the stripe android SDK, that causes the G-Pay button to be hidden for the `testEnv: true`.
Manually opening the Google Pay sheet with `Stripe.instance.confirmPlatformPayPaymentIntent(...)` works fine and shows the test cards available from google.

**Reply by captnCC** [reply]
A short addition to it: For the test mode to work (even with googles test cards) you need a valid and activated card entered in your google wallet.
After this step, the "Pay with GPay" button will appear and you can test purchases with test cards

**Reply by nayanbabariya** [reply]
@captnCC I'm from India and I've installed the Google Wallet app. I've set the region to the US. but I'm not able to see the option of adding a card to it. Could you help me?
I want to show the Google Pay button on the stripe payment sheet, integrating the pay plugin and using `PlatformPayButton` works fine.

**Comment by nizamializadeh** [comment]
Has anyone solved the problem? I have the same problem.

**Reply by nayanbabariya** [reply]
@nizamializadeh Connect your Android device with an active VPN of the country where GPay is supported. i.e US, Singapore, etc. then open the GWallet app and add the real card (No worry, your real money will not be debited). then open your payment sheet in your Flutter app.

**Comment by nizamializadeh** [comment]
@nayanbabariya  Google Pay is active in my country and I added a real card to Google Wallet, but the g_pay button does not appear in my application. Only when I set testEnv: false does the button appear, but then I get the OR_BIBED_011 error. In addition, I am testing these operations on an emulator instead of a real device. Can you help?
